src/CFBG.cpp

```diff
@@ -126,75 +126,238 @@ uint32 CFBG::GetAllPlayersCountInBG(Battleground* bg)
-            FakeRace = RACE_TAUREN;
+uint32 CFBG::getMorphFromRace(uint8 race, uint8 gender) {
+
+    if (gender == GENDER_MALE) {
+        switch(race) {
+            case RACE_ORC:
+                return FAKE_M_FEL_ORC;
+            case RACE_DWARF:
+                return FAKE_M_DWARF;
+            case RACE_NIGHTELF:
+                return FAKE_M_NIGHT_ELF;
+            case RACE_DRAENEI:
+                return FAKE_M_BROKEN_DRAENEI;
+            case RACE_TROLL:
+                return FAKE_M_TROLL;
+            case RACE_HUMAN:
+                return FAKE_M_HUMAN;
+            case RACE_BLOODELF:
+                return FAKE_M_BLOOD_ELF;
+            case RACE_GNOME:
+                return FAKE_M_GNOME;
+            case RACE_TAUREN:
+                return FAKE_M_TAUREN;
+            default:
+                return FAKE_M_BLOOD_ELF; // this should never happen, it's to fix a warning about return value
         }
-        else if (player->getGender() == GENDER_MALE) // HORDE PLAYER, ONLY HAVE MALE Night Elf ID
-        {
-            FakeMorph = FAKE_M_NELF;
-            FakeRace = RACE_NIGHTELF;
+    } else {
+        // FEMALE races DWARF, NIGHT ELF, TROLL are missing
+        switch(race) {
+            case RACE_ORC:
+                return FAKE_F_ORC;
+            case RACE_DRAENEI:
+                return FAKE_F_DRAENEI;
+            case RACE_HUMAN:
+                return FAKE_F_HUMAN;
+            case RACE_BLOODELF:
+                return FAKE_F_BLOOD_ELF;
+            case RACE_GNOME:
+                return FAKE_F_GNOME;
+            case RACE_TAUREN:
+                return FAKE_F_TAUREN;
+            default:
+                return FAKE_F_BLOOD_ELF; // this should never happen, it to fix a warning about return value
         }
-        else
-            FakeRace = player->GetTeamId(true) == TEAM_ALLIANCE ? RACE_BLOODELF : RACE_HUMAN;
+    }
+}
 
-        if (player->GetTeamId(true) == TEAM_HORDE)
-        {
-            if (player->getGender() == GENDER_MALE)
-                FakeMorph = 19723;
-            else
-                FakeMorph = 19724;
+
+void CFBG::randomRanceMorph(uint8* race, uint32* morph, TeamId team, uint8 _class, uint8 gender) {
+
+    // if alliance find a horde race
+    if (team == TEAM_ALLIANCE) {
+
+        // default race because UNDEAD morph is missing
+        *race = RACE_BLOODELF;
+
+        if (gender == GENDER_MALE) {
+            *morph = FAKE_M_BLOOD_ELF;
+
+            switch (_class) {
+                case CLASS_DRUID:
+                    *race = RACE_TAUREN;
+                    *morph = FAKE_M_TAUREN;
+                    break;
+                case CLASS_SHAMAN:
+                case CLASS_WARRIOR:
+                    *race = getRandomRace({ RACE_ORC, RACE_TAUREN, RACE_TROLL });
+                    *morph = getMorphFromRace(*race, gender);
+                    break;
+                case CLASS_PALADIN:
+                    // BLOOD ELF, so default race
```

Comment:
I'd still fill this

Reply:
I don't, because probably will take ever the default race/morph